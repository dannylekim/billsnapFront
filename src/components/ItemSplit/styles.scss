@import "../../variables.scss";

hr.card__separator {
  background-color: rgba(0, 0, 0, 0.63);
  width: 80.67%;
  margin: 1em auto;
}

.one__liner {
  display: flex;
  margin-top: 1em;
}

.grow__line {
  flex-grow: 2;
}

.modal__text {
  font-weight: 500;
  text-align: left;
  line-height: 2em;
  font-family: $primary-font;
  font-style: normal;
  color: $secondary-text-color
}

.embolden {
  font-weight: bold
}

.modal__body {
  height: 60%;
  overflow-y: auto;
}

.center {
  text-align: center;
}