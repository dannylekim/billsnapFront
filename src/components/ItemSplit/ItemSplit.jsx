import React from "react";
import {CardBody, Progress} from "shards-react";
import AccountStatusIcon from "../AccountStatusIcon";
import "./styles.scss";

function getItemSplit(itemInformation) {
  return (
    <>
      {itemInformation.accounts.map((accountItem, key) => {
        const itemCostByPercentage =
          itemInformation.cost * (accountItem.percentage / 100);
        return (
          <div key={key} className="one__liner">
            <AccountStatusIcon
              status={accountItem.status}
              name={accountItem.firstName}
            />
            <div className="grow__line">
              <div className="center">{itemCostByPercentage.toFixed(2)} $</div>
              <Progress theme="success" value={accountItem.percentage}>
                {accountItem.percentage} %
              </Progress>
            </div>
          </div>
        );
      })}
    </>
  );
}

const ItemSplit = ({ itemInformation }) => {
  return (
    <CardBody className="modal__text modal__body">
      <div>
        Item: <span className="embolden"> {itemInformation.name} </span>
      </div>
      <div>Total: {itemInformation.cost} $</div>
      <hr className="card__separator" />
      <div>Shared between:</div>
      {getItemSplit(itemInformation)}
    </CardBody>
  );
};

export default ItemSplit;
