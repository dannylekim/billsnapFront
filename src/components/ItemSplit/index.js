import ItemSplit from "./ItemSplit";

export default ItemSplit;