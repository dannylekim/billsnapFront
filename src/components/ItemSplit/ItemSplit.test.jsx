import React from "react";
import ItemSplit from "./ItemSplit";

const testItemInformation = {
  name: "Bigger String",
  cost: "300.00",
  accounts: [
    { status: "ACCEPTED", firstName: "string", percentage: 60 },
    { status: "ACCEPTED", firstName: "string", percentage: 40 },
  ],
};

describe("ItemSplit", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("should match snapshot", () => {
        matches(<ItemSplit itemInformation={testItemInformation} />);
      });
    });
  });
});
