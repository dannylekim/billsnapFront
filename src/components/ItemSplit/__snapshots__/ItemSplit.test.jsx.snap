// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSplit render snapshots ðŸ“¸ should match snapshot 1`] = `
<div
  className="modal__text modal__body card-body"
>
  <div>
    Item: 
    <span
      className="embolden"
    >
       
      Bigger String
       
    </span>
  </div>
  <div>
    Total: 
    300.00
     $
  </div>
  <hr
    className="card__separator"
  />
  <div>
    Shared between:
  </div>
  <div
    className="one__liner"
  >
    <div>
      <div
        className="accounts__icons"
      >
        <svg
          fill="currentColor"
          height={50}
          size={50}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
        <svg
          className="status_icon"
          color="#47E5B6"
          fill="currentColor"
          height={40}
          size={40}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#47E5B6",
            }
          }
          viewBox="0 0 24 24"
          width={40}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </div>
      <p
        className="icon__name"
      >
        string
      </p>
    </div>
    <div
      className="grow__line"
    >
      <div
        className="center"
      >
        180.00
         $
      </div>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin="0"
          aria-valuenow={60}
          className="progress-bar bg-success"
          role="progressbar"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          60
           %
        </div>
      </div>
    </div>
  </div>
  <div
    className="one__liner"
  >
    <div>
      <div
        className="accounts__icons"
      >
        <svg
          fill="currentColor"
          height={50}
          size={50}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
        <svg
          className="status_icon"
          color="#47E5B6"
          fill="currentColor"
          height={40}
          size={40}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#47E5B6",
            }
          }
          viewBox="0 0 24 24"
          width={40}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </div>
      <p
        className="icon__name"
      >
        string
      </p>
    </div>
    <div
      className="grow__line"
    >
      <div
        className="center"
      >
        120.00
         $
      </div>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin="0"
          aria-valuenow={40}
          className="progress-bar bg-success"
          role="progressbar"
          style={
            Object {
              "width": "40%",
            }
          }
        >
          40
           %
        </div>
      </div>
    </div>
  </div>
</div>
`;
